src/TuxMutableRenderMixin.js

```diff
@@ -0,0 +1,131 @@
+// @param mutableTrait STRING trait to find the path of keys to within the searchObject
+// @param searchObject OBJECT:
+  // expected keys ANY: this object can contain any keys, but is expected to contain the key specified in mutableTrait
+// @param currentPath ARRAY: array of strings that lead to the specified mutableTrait within searchObject
+// @param mutablePath ARRAY (CLOSURE from deepSearch): array that will contain the first viable path in the searchObject
+var findPath = function (mutableTrait, searchObject, currentPath) {
+  // if the searchObject is undefined or null, return as the path isn't down this branch
+  if (searchObject === undefined || searchObject === null) {
+    return;
+  } else if (searchObject.hasOwnProperty(mutableTrait)) {
+    // Add mutableTrait to the end of the currentPath and set it to mutablePath and return
+    mutablePath = currentPath.concat(mutableTrait);
+    return;
+  }
+
+  // declare reused variables
+  var key, newPath;
+
+  for (key in searchObject) {
+    // add the new key onto currentPath
+    newPath = currentPath.concat(key);
+    // recurse over the item at searchObject[key] with newPath as the currentPath input
+    findPath(searchObject[key], newPath);
+    // break out of loop if a single valid path has been found
+    if (mutablePath.length > 0) {
+      break;
+    }
+  }
+};
+
+// deepSearch FUNCTION: creates an array of string keys to transverse the object specified by the @param type key. Currently matches only the first path found.
+// @param mutableTrait STRING: trait to find the path of keys to within the objectToSearch
+// @param objectToSearch OBJECT:
+  // expected keys ANY: this object can contain any keys, but is expected to contain the key specified in mutableTrait
+// @param type STRING: either the string 'props' or 'state'
+var deepSearch = function (mutableTrait, objectToSearch, type) {
+  var mutablePath = [];
+  var properType = type === 'props' || type === 'state';
+
+  invariant(properType, 'Input param "type" needs to be either \'props\' or \'state\'')
+
+  // call recursive function findPath and pass in type as an array to match expected input type (array)
+  findPath(objectToSearch, [type], mutableTrait);
+  return mutablePath;
+};
+
+// TuxMutableRenderMixin OBJECT: adds mixin to React class that will add componentWillMount and shouldComponentUpdate life cycle events to the component
+module.exports = {
+  // componentWillMount FUNCTION: creates an array of string key paths attached to the component's constructor with a key of __tuxMutableTraits__.
+  // These paths are used in the shouldComponentUpdate life cycle event to compare current and next values of the specified traits to determine
+  // if the component should re-render.
+  componentWillMount: function () {
+    var mutableTraits = this.mutableTraits;
+    var mutableTraitsPaths = [];
+    var type, isArray, isString, mutableTraitsLength, i;
+
+    if (this.constructor.__tuxMutableTraits__) {
```

Comment:
Change to `if (!this.constructore.__tuxMutableTraits__) {...}` instead of `if (this.constructore.__tuxMutableTraits__) { return; } else {...}` as it is more semantic.